### PR TITLE
Update CDI thirdparty FAT and run on EE10

### PR DESCRIPTION
**Pull request by Azquelt**
For #19867

### CODE REVIEW

File: dev/com.ibm.ws.cdi.third.party_fat/fat/src/com/ibm/ws/cdi/thirdparty/tests/HibernateSearchTest.java [hunk 1]
```diff
@@ -57,24 +58,23 @@
 
 */
 
+@SkipForRepeat(EERepeatActions.EE7_ID)
 @RunWith(FATRunner.class)
 public class HibernateSearchTest extends FATServletClient {
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.third.party_fat/fat/src/com/ibm/ws/cdi/thirdparty/tests/HibernateSearchTest.java:65]: This test never ran on EE7.

The old configuration said "run in the original config, then EE8, then EE9". However, the server.xml for this server was set up for EE8 which meant the test ran on EE8 twice.

File: dev/com.ibm.ws.cdi.third.party_fat/fat/src/com/ibm/ws/cdi/thirdparty/tests/ThirdPartyTests.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import componenttest.topology.utils.HttpUtils;
+
+@RunWith(FATRunner.class)
+public class ThirdPartyTests extends FATServletClient {
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.third.party_fat/fat/src/com/ibm/ws/cdi/thirdparty/tests/ThirdPartyTests.java:45]: Previously this test was marked `@Mode(FULL)`. However, as it wasn't annotated with `@RunWith(FATRunner)`, the `@Mode` annotation was ignored.

File: dev/com.ibm.ws.cdi.third.party_fat/fat/src/com/ibm/ws/cdi/thirdparty/tests/FATSuite.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                ThirdPartyTests.class
+})
+public class FATSuite {
+
+    @ClassRule
+    public static RepeatTests r = EERepeatActions.repeat(null,
+                                                         EERepeatActions.EE8,
+                                                         EERepeatActions.EE10,
+                                                         EERepeatActions.EE9,
+                                                         EERepeatActions.EE7);
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.third.party_fat/fat/src/com/ibm/ws/cdi/thirdparty/tests/FATSuite.java:36]: The old config ran every repeat in LITE mode. I've changed it to only run one.

**Comment by Azquelt** [comment]
#build
#libby

**Review by benjamin-confino — APPROVED** [review]

**Comment by Azquelt** [comment]
Part of the build did not run due to an infrastructure problem but all the tests ran and passed.

**Pull request merged**